pkg/certs/certs.go

```diff
@@ -279,6 +287,7 @@ func (pair *KeyPair) RenewCertificate(caPrivateKey *ecdsa.PrivateKey, parentCert
 	newCertificate.NotBefore = notBefore
 	newCertificate.NotAfter = notAfter
 	newCertificate.SerialNumber = serialNumber
+	newCertificate.DNSNames = altDNSNames
```

Comment:
FYI: here I've switched to always set the altDNSNames as I feel it's better to have a declarative approach here, if one passes nil, the generated Certificate must have no DNS names, even if it used to have some previously. I feel it could cause less issues in the future, but I'd like to have some other colleague opinion here.